src/app/core/header/timepiece/timepiece.controller.test.ts

```diff
@@ -0,0 +1,112 @@
+import { TimepieceComponent } from './timepiece.component';
+
+describe('Timepiece', () => {
+    let sut;
+
+    beforeEach(() => {
+        sut = new TimepieceComponent();
+    });
+
+    describe('On Init', () => {
+
+        beforeEach(() => {
+            spyOn(sut, 'createTimer');
+            spyOn(sut, 'subscribeToTimer');
+            sut.ngOnInit();
+        });
+
+        it('should create the timer', () => {
+            expect(sut.createTimer).toHaveBeenCalled();
+        });
+
+        it('should subscribe to timer', () => {
+            expect(sut.subscribeToTimer).toHaveBeenCalled();
+        });
+
+        it('timer should be undefined', () => {
+            expect(sut.timer).toBeUndefined();
+        });
+
+        it('timeTracker should be undefined', () => {
+            expect(sut.timeTracker).toBeUndefined();
+        });
+
+    });
+
+    describe('On timer create', () => {
+
+        beforeEach(() => {
+            sut.createTimer(0, 1000);
```

Comment:
Any value can be passed to this function. I think it si not good idea to pass fixed value in test. Better to pass:
`sut.createTimer(mockDueTime, mockPeriod);`